tutorials/intro/Intro_Tutorial_1.ipynb

```diff
@@ -144,41 +195,208 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining a `Candidate` schema\n",
+    "We now define the schema of the relation mention we want to extract (which is also the schema of the candidates).  This must be a subclass of `Candidate`, and we define it using a helper function. Here we'll define a binary _spouse relation mention_ which connects two `Span` objects of text.  Note that this function will create the table in the database backend if it does not exist:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from snorkel.models import candidate_subclass\n",
+    "\n",
+    "Spouse = candidate_subclass('Spouse', ['person1', 'person2'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Writing a basic `CandidateExtractor`\n",
+    "\n",
+    "Next, we'll write a basic function to extract **candidate spouse relation mentions** from the corpus.  The [Spacy](https://spacy.io/) parser we used performs _named entity recognition_ for us.\n",
+    "\n",
+    "We will extract `Candidate` objects of the `Spouse` type by identifying, for each `Sentence`, all pairs of ngrams (up to trigrams) that were tagged as people.  We do this with three objects:\n",
+    "\n",
+    "* A `ContextSpace` defines the \"space\" of all candidates we even potentially consider; in this case we use the `Ngrams` subclass, and look for all n-grams up to 3 words long\n",
+    "\n",
+    "* A `Matcher` heuristically filters the candidates we use.  In this case, we just use a pre-defined matcher which looks for all n-grams tagged by CoreNLP as \"PERSON\"\n",
+    "\n",
+    "* A `CandidateExtractor` combines this all together!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from snorkel.candidates import Ngrams, CandidateExtractor\n",
+    "from snorkel.matchers import PersonMatcher\n",
+    "\n",
+    "ngrams         = Ngrams(n_max=7)\n",
+    "person_matcher = PersonMatcher(longest_match_only=True)\n",
+    "cand_extractor = CandidateExtractor(Spouse, \n",
+    "                                    [ngrams, ngrams], [person_matcher, person_matcher],\n",
+    "                                    symmetric_relations=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we'll split up the documents into train, development, and test splits; and collect the associated sentences.\n",
+    "\n",
+    "Note that we'll filter out a few sentences that mention more than five people. These lists are unlikely to contain spouses."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
```

Comment:
That's fine. (Doesn't bother me though, just a bit of Python :D)

I'd suggest a separate helper that takes an optional filter function as an argument.